File: app/src/main/java/com/aicloudflare/musicbox/Song.java
package com.aicloudflare.musicbox;

public class Song {
    private String title;
    private String artist;
    private int coverResId;
    private boolean isAdded; // Biến quan trọng để xác định icon

    public Song(String title, String artist, int coverResId, boolean isAdded) {
        this.title = title;
        this.artist = artist;
        this.coverResId = coverResId;
        this.isAdded = isAdded;
    }

    public String getTitle() { return title; }
    public String getArtist() { return artist; }
    public int getCoverResId() { return coverResId; }
    public boolean isAdded() { return isAdded; }
    public void setAdded(boolean added) { isAdded = added; }
}

File: app/src/main/java/com/aicloudflare/musicbox/model/Song.kt
package com.aicloudflare.musicbox.data.model

data class Song(
    val id: String = "",
    val title: String = "",
    val artist: String = "",
    val album: String = "",
    val duration: Long = 0L,
    val albumArtUrl: String = "",
    // Quan trọng cho Người C (ExoPlayer)
    val audioUrl: String = "",
    // Quan trọng cho thống kê Tải nhạc/Thịnh hành
    val downloadCount: Long = 0L
)